From 8a3b76b42dde08b2ba6ba2e5a3d736f621effca3 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 18 Jul 2022 14:26:11 -0400
Subject: [PATCH] windowManager.js: Fix workspace switching when
 only-on-primary is enabled.

The monitor index needs to be passed, not the monitor object.

fixes: #10943
---
 js/ui/windowManager.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index 49f2270920..9eeed2aa9b 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -1204,7 +1204,7 @@ var WindowManager = class WindowManager {
         if (global.settings.get_boolean('workspace-osd-visible')) {
             let current_workspace_index = global.screen.get_active_workspace_index();
             if (this.wm_settings.get_boolean('workspaces-only-on-primary')) {
-                this._showWorkspaceOSDOnMonitor(Main.layoutManager.primaryMonitor, current_workspace_index);
+                this._showWorkspaceOSDOnMonitor(Main.layoutManager.primaryMonitor.index, current_workspace_index);
             }
             else {
                 let {monitors} = Main.layoutManager;
